copied trivial from cats.core

kailuowang opened the pull request.

milessabin commented on core/src/main/scala/cats/derived/trivial.scala:
```diff
@@ -20,3 +20,27 @@ trait Trivial1[F[_]]
+sealed trait Trivial
+
+object Trivial {
+  type P1[A] = Trivial
```
What's the motivation for these aliases?

milessabin commented on core/src/main/scala/cats/derived/trivial.scala:
```diff
@@ -20,3 +20,27 @@ trait Trivial1[F[_]]
+  type P3[A, B, C] = Trivial
+  type P3H1[F[_], A, B, C] = Trivial
+
+  implicit val catsTrivialInstance: Trivial = new Trivial {}
```
I think `mkTrivial` would be better here, for consistency with `Trivial1`.

milessabin commented:
OK, looking more closely, `Trivial.PH1` replaces `Trivial1` ... can we motivate the naming of the aliases and ditch `Trivial1` everywhere?

kailuowang commented:
I replaced Trivial1 and renamed the instance creation method. By "motivate the naming of the aliases", did you mean naming them in a way that shows motivation?

milessabin commented:
Well I'd like to know what the current set of names are meant to convey ... there's a pattern there, but it's not clear what it is or what the choice of letters signifies.

kailuowang commented:
I had the same feeling, but I struggle to come up with a naming scheme myself. pinging @djspiewak

djspiewak commented:
I'm guessing I meant "partial higher-order position arity-1", but that's just a guess. If the alias isn't obvious, it's a bad alias.

kailuowang commented:
Closing due to lack of interest
